pkg/registry/deployment/strategy.go

```diff
@@ -92,6 +92,8 @@ func (deploymentStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newDeployment := obj.(*extensions.Deployment)
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Spec = oldDeployment.Spec
+	newDeployment.Labels = oldDeployment.Labels
```

Comment:
@kargakis @janetkuo I understand this is status update.

We support updating annotations via the /binding subresource. There are use cases for controllers to update annotations along with status, though we don't do it now.

I agree that we shouldn't allow labels to be changed via status updates. @kargakis Please file an issue to fix the other status endpoints. It is technically a breaking API change, but one could view it as a vulnerability, so we should fix it.